examples/dnsblock/dnsblock.lua

```diff
@@ -0,0 +1,74 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- Filter DNS packets based on a blocklist
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local blacklist = {
+	"github.com",
+	"gitlab.com",
+}
+
+local function nop() end
+
+local function check_blacklist(name)
+	for _, v in ipairs(blacklist) do
+		if string.find(name, v) ~= nil then
+			return true
+		end
+	end
+	return false
+end
+
+local function get_domainname(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until (d == 0)
+
+	return name .. skb:getstring(off, i)
+end
+
+local function dnsblock_mt(skb, par)
+	-- ip header
```

Comment:
what you mean here?

Reply:
Ah, that should not be here, it was there earlier, when I was manually extracting the udp offset. Now we use `thoff`